src/states/elder/mod.rs

```diff
@@ -717,16 +717,22 @@ impl Elder {
         let dst = msg.message_dst();
         let signed_msg = msg.signed_routing_message();
         if self.in_location(dst) {
-            self.check_signed_message_trust(signed_msg)?;
+            let status = self.check_signed_message_trust(signed_msg)?;
             self.check_signed_message_integrity(signed_msg)?;
-            self.update_our_knowledge(signed_msg);
 
             if dst.is_multiple() {
```

Comment:
I've been thinking fora while that this should be done before checking the trust or integrity, would it make sense to swap it now, to avoid having the result then check status later?